providers/gce/gce_util.go

```diff
@@ -389,3 +390,17 @@ func removeString(slice []string, s string) []string {
 	}
 	return newSlice
 }
+
+// GetStaticIPFromLoadBalancer get Public IP from the load balancer
+func GetStaticIPFromLoadBalancer(svc *v1.Service) string {
+
+	if ip, ok := svc.Annotations[ServiceAnnotationLBIPv4]; ok && ip != "" {
+		return ip
+	}
+
+	if v := reflect.ValueOf(&svc.Spec).Elem().FieldByName("LoadBalancerIP"); v.IsValid() {
```

Comment:
Is the reflection here because the field does not exist on the version of client-go currently impoirted?

Reply:
the field was deprecated but it will not be removed, I can't understand the reflection neither

Reply:
@aojea  
please see the below deprecation logs from Kubernetes 1.24, field `LoadBalancerIP` may be removed. for now, this field is still alive. 
```
Deprecated Service.Spec.LoadBalancerIP. This field was under-specified and its meaning varies across implementations. As of Kubernetes v1.24, users are encouraged to use implementation-specific annotations when available. This field may be removed in a future API version. (https://github.com/kubernetes/kubernetes/pull/107235, [@uablrek](https://github.com/uablrek))
```

@jpbetz yes, from my perspective, if there is a conditional statement to check whether `LoadBalancerIP` exists, before using it. it will improve flexibility and stability.

I'm not insisting on this point but advise on it.

Reply:
I have doubts we are going to remove a field from a GA API,

Reply:
@aojea em, update with using the `LoadBalancerIP` directly.

Reply:
Yes, use `LoadBalancerIP` directly.